webapp/channels/src/components/admin_console/access_control/editors/cel_editor/editor.tsx

```diff
@@ -0,0 +1,363 @@
+
+        monacoRef.current.getModel()?.onDidChangeContent(() => {
+            const newValue = monacoRef.current?.getValue() || '';
+            handleChange(newValue);
+        });
+
+        monacoRef.current.onDidChangeCursorPosition((e) => {
+            setEditorState((prev) => ({
+                ...prev,
+                cursorPosition: {line: e.position.lineNumber, column: e.position.column},
+            }));
+        });
+
+        // To disable monaco's default behavior of opening the find and replace widget
+        monaco.editor.addKeybindingRule({
+            keybinding: monaco.KeyMod.CtrlCmd | monaco.KeyCode.KeyF,
+            command: null,
+        });
+
+        monaco.editor.addCommand({
+            id: VALIDATE_POLICY_SYNTAX_COMMAND_ID,
+            run: validateSyntax,
+        });
+
+        monaco.editor.addKeybindingRule({
+            keybinding: monaco.KeyMod.Alt | monaco.KeyCode.Enter,
+            command: VALIDATE_POLICY_SYNTAX_COMMAND_ID,
+        });
+
+        return () => {
+            if (monacoRef.current) {
+                monacoRef.current.dispose();
+                monacoRef.current = null;
+            }
+        };
+    }, []);
+
+    return (
+        <div className={`cel-editor ${className}`}>
+            <MonacoLanguageProvider schemas={schemas}/>
+
+            <div
+                className='cel-editor__container'
+                data-status-color={editorState.statusBarColor}
+            >
+                {!editorState.expression && (
+                    <div
+                        className='policy-editor-placeholder'
+                        aria-label='CEL Expression Editor'
+                    >
+                        {placeholder}
+                    </div>
+                )}
+
+                <div
+                    ref={editorRef}
+                    className='cel-editor__input'
+                />
+                <div
+                    className='cel-editor__status-bar'
+                    style={{backgroundColor: editorState.statusBarColor}}
+                    onClick={() => {
+                        if (!editorState.isValidating && editorState.validationErrors.length === 0 &&
+                            !(editorState.isValid && editorState.statusBarColor === 'var(--online-indicator)')) {
+                            validateSyntax();
+                        }
+                    }}
+                    role='button'
+                    tabIndex={0}
+                    onKeyDown={(e) => {
+                        if (e.key === 'Enter' || e.key === ' ') {
+                            if (!editorState.isValidating && editorState.validationErrors.length === 0 &&
+                                !(editorState.isValid && editorState.statusBarColor === 'var(--online-indicator)')) {
+                                validateSyntax();
+                            }
+                        }
+                    }}
+                    data-validation-state={
```

Comment:
(Post-merge) When you change `statusBarColor` to instead be an enum based on the other state, could you also remove this and replace it with a CSS class? While you can technically use data attributes for this, that's not something you'd typically do when using React